File: app/src/main/java/com/inspiringteam/xchange/data/source/QuakesDataSource.java
package com.inspiringteam.xchange.data.source;

import androidx.annotation.NonNull;

import com.inspiringteam.xchange.data.models.Quake;

import java.util.List;

import io.reactivex.Single;


public interface QuakesDataSource {
    @NonNull
    Single<List<Quake>> getQuakes();

    @NonNull
    Single<Quake> getQuake(@NonNull String quakeId);

    void saveQuakes(@NonNull List<Quake> quakes);

    void saveQuake(@NonNull Quake quake);

    void deleteAllQuakes();

    void deleteQuake(@NonNull String quakeId);
}

File: app/src/main/java/com/inspiringteam/xchange/data/source/QuakesDataSource.kt
package com.inspiringteam.xchange.data.source

import com.inspiringteam.xchange.data.models.Quake
import io.reactivex.Single

interface QuakesDataSource {
    fun getQuakes(): Single<List<Quake>>
    fun getQuake(quakeId: String): Single<Quake>
    fun saveQuakes(quakes: List<Quake>)
    fun saveQuake(quake: Quake)
    fun deleteAllQuakes()
    fun deleteQuake(quakeId: String)
}